src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -167,6 +167,20 @@ namespace cryptonote
 
     tx.extra = extra;
     keypair txkey = keypair::generate();
+
+    // the back of destinations is assumed to be the change address (or the dummy when sweeping)
```

Comment:
OK, then I'll just pass the change address as well to `construct_tx_and_get_tx_key` so that any ambiguity is avoided.